samza-api/src/main/java/org/apache/samza/context/SamzaContext.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.samza.context;
+
+/**
+ * Container object for all context provided by the Samza framework to instantiate an application at runtime.
+ */
+public interface SamzaContext {
+  /**
+   * @return framework-provided context for the overall job that is being run
+   */
+  JobContext getJobContext();
+
+  /**
+   * @return framework-provided context for the container that this is in
+   */
+  ContainerContext getContainerContext();
```

Comment:
Can you please share the rationale behind having three different wrapper classes : `SamzaContainerContext`, `Context`, `ContainerContext` each one of them housing some of the context objects. In what scenarios or API each one of these individual classes will be used .

Reply:
SamzaContainerContext has been removed.
Context holds all of the context objects (e.g. job/container/task context, app-defined context). ContainerContext is just for framework-provided context at the container level (e.g. container metrics, container id).